src/Simplify.cpp

```diff
@@ -3908,6 +3924,68 @@ class Simplify : public IRMutator {
             IRMutator::visit(op);
         }
     }
+    template <typename T>
+    Expr hoist_slice_vector(Expr e) {
```

Comment:
You can pass the args as a, b, and still templatize the function on T so that you can call T::make.